analyzers/src/SonarAnalyzer.CSharp/Rules/UseUriInsteadOfString.cs

```diff
@@ -121,85 +121,74 @@ private static void VerifyPropertyDeclaration(SyntaxNodeAnalysisContext context)
         private static void VerifyRecordDeclaration(SyntaxNodeAnalysisContext context)
         {
             var declaration = (RecordDeclarationSyntaxWrapper)context.Node;
-
-            if (context.ContainingSymbol.Kind == SymbolKind.NamedType
-                && HasStringUriParams(declaration.ParameterList, context.SemanticModel))
+            if (!context.IsRedundantPositionalRecordContext() && HasStringUriParams(declaration.ParameterList, context.SemanticModel))
             {
                 context.ReportIssue(Diagnostic.Create(RuleS3996, declaration.SyntaxNode.GetLocation()));
             }
         }
 
         private static bool HasStringUriParams(BaseParameterListSyntax parameterList, SemanticModel model) =>
             parameterList != null
-            && parameterList.Parameters.Any(parameter => NameContainsUri(parameter.Identifier.Text)
-                                                         && model.GetDeclaredSymbol(parameter).IsType(KnownType.System_String));
+            && parameterList.Parameters.Any(x => NameContainsUri(x.Identifier.Text) && model.GetDeclaredSymbol(x).IsType(KnownType.System_String));
 
         private static void VerifyInvocationAndCreation(SyntaxNodeAnalysisContext context)
         {
-            if (!(context.SemanticModel.GetSymbolInfo(context.Node).Symbol is IMethodSymbol invokedMethodSymbol)
-                || invokedMethodSymbol.IsInType(KnownType.System_Uri))
-            {
-                return;
-            }
-
-            var stringUrlParams = GetStringUrlParamIndexes(invokedMethodSymbol);
-            var methodOverloads = FindOverloadsThatUseUriTypeInPlaceOfString(invokedMethodSymbol, stringUrlParams);
-            if (stringUrlParams.Count > 0 && methodOverloads.Any())
+            if (context.SemanticModel.GetSymbolInfo(context.Node).Symbol is IMethodSymbol invokedMethodSymbol && !invokedMethodSymbol.IsInType(KnownType.System_Uri))
```

Comment:
Why do we never pass the cancelation token?
```suggestion
            if (context.SemanticModel.GetSymbolInfo(context.Node, context.CancellationToken).Symbol is IMethodSymbol invokedMethodSymbol && !invokedMethodSymbol.IsInType(KnownType.System_Uri))
```

Reply:
That's a good question. I don't know 😄 We just don't. Please add a point to `How we work`.